src/classes/Contacts.cls

```diff
@@ -245,4 +379,537 @@ public inherited sharing class Contacts extends fflib_SObjects {
         return oppIds;
     }
 
+    public void resetAddresses() {
+        Map<Id, Address__c> mapAccIdAddr = new Map<Id, Address__c>();
+        for (Contact con : (List<Contact>) getRecords()) {
+            mapAccIdAddr.put(con.AccountId, null);
+        }
+
+        // get the default and seasonal address for each HH
+        Map<Id, Addresses.HHInfo> mapAccIdHHInfo =
+                Addresses.refreshCurrentHHAddress(mapAccIdAddr, null, true);
+
+        // reset the Contact's address fields with the appropriate address
+        for (Contact con : (List<Contact>) getRecords()) {
+            Addresses.HHInfo hhinfo = mapAccIdHHInfo.get(con.AccountId);
+            if (hhinfo != null) {
+                Address__c addr = hhinfo.addrSeasonal;
+                if (addr == null) {
+                    addr = hhinfo.addrDefault;
+                }
+                if (addr != null) {
+                    NPSP_Address npspAddress = new NPSP_Address(addr);
+                    NPSP_Contact npspContact = new NPSP_Contact(con);
+                    npspContact.currentAddress(npspAddress);
+                }
+            }
+        }
+    }
+
+    public List<NPSP_Contact> npspContacts(){
+        List<NPSP_Contact> npspContacts = new List<NPSP_Contact>();
+        for (Contact so : (List<Contact>)getRecords()) {
+            NPSP_Contact npspContact = new NPSP_Contact(so, oldVersionOf(so));
+            npspContacts.add(npspContact);
+        }
+        return npspContacts;
+    }
+
+    public Contacts contactsInAccountsOfTypeHousehold() {
+        List<Contact> contactsInHouseholdAccounts = new List<Contact>();
+        for (SObject so : getRecords()) {
+            NPSP_Contact npspContact = new NPSP_Contact((Contact) so);
+            if (isAnAccountOfTypeHousehold(accountFor(npspContact))) {
+                contactsInHouseholdAccounts.add(npspContact.getRecord());
+            }
+        }
+        return new Contacts(contactsInHouseholdAccounts);
+    }
+
+    private Account accountFor(NPSP_Contact npspContact) {
+        return accountsByIds().get(npspContact.accountId());
+    }
+
+    public Contacts thatHaveAccounts() {
```

Comment:
```suggestion
    public Contacts getContactsWithAccount() {
```

Reply:
This is a filter method that returns a subset of the underlying records.  Its usage is 
`Contacts consInAccts = contacts.thatHaveAccounts();`.  

`getContactsWithAccount()` sounds to me more like a query method that would return all of the underlying records with their parent Account record.